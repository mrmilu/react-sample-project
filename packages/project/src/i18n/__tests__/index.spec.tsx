import { render } from '@testing-library/react';
import React, { useEffect } from 'react';
import { Locale, useTranslation } from '../';

type MockComponentProps = {
  lang?: Locale;
};

function MockComponent({ lang = 'en' }: MockComponentProps) {
  const { t, changeLanguage } = useTranslation();
  useEffect(() => {
    async function setLanguage() {
      await changeLanguage(lang);
    }

    void setLanguage();
  }, [changeLanguage, lang]);
  return <div>{t('title')}</div>;
}

describe('i18n', () => {
  it('Should translate a text in english', () => {
    const renderer = render(<MockComponent />);
    expect(renderer.getByText('Sample title')).not.toBeNil();
    expect(renderer.container).toMatchSnapshot();
  });
  it('Should translate a text in spanish', () => {
    const renderer = render(<MockComponent lang="es" />);
    expect(renderer.getByText('TÃ­tulo de ejemplo')).not.toBeNil();
    expect(renderer.container).toMatchSnapshot();
  });
});
