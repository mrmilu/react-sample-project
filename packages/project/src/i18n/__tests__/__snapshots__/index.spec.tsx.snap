// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`i18n Should translate a text in english 1`] = `
<div>
  <div>
    Sample title
  </div>
</div>
`;

exports[`i18n Should translate a text in spanish 1`] = `
<div>
  <div>
    TÃ­tulo de ejemplo
  </div>
</div>
`;
